sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/ConfidentialLedgerClient.cs

```diff
@@ -19,55 +18,44 @@ public partial class ConfidentialLedgerClient
         /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
         /// <param name="options"> The options for configuring the client. </param>
         public ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential, ConfidentialLedgerClientOptions options)
-        {
-            if (ledgerUri == null)
-            {
-                throw new ArgumentNullException(nameof(ledgerUri));
-            }
-            if (credential == null)
-            {
-                throw new ArgumentNullException(nameof(credential));
-            }
-
-            var actualOptions = options ?? new ConfidentialLedgerClientOptions();
-            var transportOptions = GetIdentityServerTlsCertAndTrust(ledgerUri, actualOptions);
-            ClientDiagnostics = new ClientDiagnostics(actualOptions, true);
-            _tokenCredential = credential;
-            var authPolicy = new BearerTokenAuthenticationPolicy(_tokenCredential, AuthorizationScopes);
-            _pipeline = HttpPipelineBuilder.Build(
-                actualOptions,
-                Array.Empty<HttpPipelinePolicy>(),
-                new HttpPipelinePolicy[] { authPolicy },
-                transportOptions,
-                new ResponseClassifier());
-            _ledgerUri = ledgerUri;
-            _apiVersion = actualOptions.Version;
-        }
+            : this(ledgerUri, credential: credential, options: options, identityServiceClient: default)
+        { }
 
-         /// <summary> Initializes a new instance of ConfidentialLedgerClient. </summary>
+        /// <summary> Initializes a new instance of ConfidentialLedgerClient. </summary>
         /// <param name="ledgerUri"> The Confidential Ledger URL, for example https://contoso.confidentialledger.azure.com. </param>
-        /// <param name="certificateCredential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="clientCertificate"> A <see cref="X509Certificate2"/> used to authenticate to an Azure Service. </param>
         /// <param name="options"> The options for configuring the client. </param>
-        public ConfidentialLedgerClient(Uri ledgerUri, CertificateCredential certificateCredential, ConfidentialLedgerClientOptions options = null)
+        public ConfidentialLedgerClient(Uri ledgerUri, X509Certificate2 clientCertificate, ConfidentialLedgerClientOptions options = null)
+            : this(ledgerUri, clientCertificate: clientCertificate, options: options, identityServiceClient: null)
+        { }
+
+        internal ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential = null, X509Certificate2 clientCertificate = null, ConfidentialLedgerClientOptions options = null, ConfidentialLedgerIdentityServiceClient identityServiceClient = null)
```

Comment:
At least from this PR - even expanding unmodified code lines in this area - when is `identityServiceClient` non-null? I see further below you even use `??` on it and create one, but haven't yet seen where this can be anything but `null`. Seems some of this code could be simplified.